### PR TITLE
Installer state controller

### PR DESCRIPTION
**Pull request by CarlosNihelton**
- Pure Dart streams controller.
- Reacts to a Stream<String> meant to be stdin.
- Will be used to parse text output from the launcher process.

### CODE REVIEW

File: lib/installer_state_controller.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+import 'dart:async';
+
+enum InstallerState {
+  kInitializing,
+  kUnpacking,
+  kSettingUp,
+  kRunning,
+  kDone,
+  kError
```
Comment (jpnurmi) [inline lib/installer_state_controller.dart:26]: the k-prefix is generally used for constants :)

File: lib/installer_state_controller.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+/// Models the WSL launcher app state change as a function of the input String.
+/// This line is meant to come from a transformation of [stdin], thus instead
+/// of calls to `add()` an event, as regular BLoC does, this controller accepts
+/// a [inputStream] in its constructor as a source of events and wraps it with
+/// an internal [StreamController].
+class InstallerStateController {
+  late StreamController<String> _eventsController;
+  late StreamController<InstallerState> _stateController;
+  late StreamSubscription<String> _eventSubs;
```
Comment (jpnurmi) [inline lib/installer_state_controller.dart:37]: all three could be `late final`

File: pubspec.yaml [hunk 3]
```diff
@@ -32,6 +32,7 @@ dependencies:
   flutter_localizations:
     sdk: flutter
   intl: ^0.17.0
+  provider: ^6.0.2
```
Comment (jpnurmi) [inline pubspec.yaml:35]: I'm sure it will be useful but not used anywhere in this PR? :)

File: test/installer_state_controller_test.dart [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ *
+ */
+
+import 'dart:async';
+
+import 'package:test/test.dart';
+import 'package:ubuntu_wsl_splash/installer_state_controller.dart';
+
+void main() {
+  group('Intallation status controller tests', () {
```
Comment (jpnurmi) [inline test/installer_state_controller_test.dart:24]: typo "intallation"

File: test/installer_state_controller_test.dart [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      ];
+
+      InstallerStateController controller = InstallerStateController(
+        Stream.fromIterable(stdoutMessages),
+        InstallerState.kInitializing,
+      );
+
+      // When:
+      // empty because this controller is meant to react upon the stream of
+      // inputs, istead of acting in a controlled call fashion.
```
Comment (jpnurmi) [inline test/installer_state_controller_test.dart:48]: typo "istead"

**Review by jpnurmi — COMMENTED** [review]
Looks good, I have only some nit-pick

File: pubspec.yaml (hunk 3 above)
Comment (CarlosNihelton) [reply pubspec.yaml:35]: Indeed. Let's take it out for now.

File: lib/installer_state_controller.dart (hunk 2 above)
Comment (CarlosNihelton) [reply lib/installer_state_controller.dart:37]: Indeed. Good catch.

File: lib/installer_state_controller.dart (hunk 1 above)
Comment (CarlosNihelton) [reply lib/installer_state_controller.dart]: I wanted to name the enum values with upper case initials, but it is not idiomatic so I cheated the linter 😅 
Fixed to the idiomatic way.

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

**Pull request merged**